test/integration/adoption_application_reviews_test.rb

```diff
@@ -143,4 +143,43 @@ class AdoptionApplicationReviewsTest < ActionDispatch::IntegrationTest
         params: { adopter_account_id: @adopter_account_id, dog_id: @dog.id }
     end
   end
+
+  test "an adopter can withdraw and remove an application and a staff can subsequently set the adoption application from withdrawn to another status and the application reappears on the adopter's applications page" do
```

Comment:
This test was actually covered in `adopter_applications_page_test.rb` "**Staff can revert withdraw and remove by an adopter and the application reappears for adopter**" but I think it is better placed here. So let's use this one and I can remove the other one. I may suggest using this shorter test name, though?

Reply:
Haha yea this shorter name is much better, I'll change it